opensearch/src/main/java/org/opensearch/sql/opensearch/request/OpenSearchQueryRequest.java

```diff
@@ -189,6 +189,9 @@ public OpenSearchResponse searchWithPIT(Function<SearchRequest, SearchResponse>
       // Set sort field for search_after
       if (this.sourceBuilder.sorts() == null) {
         this.sourceBuilder.sort(DOC_FIELD_NAME, ASC);
+        // Workaround to preserve sort location more exactly,
+        // see https://github.com/opensearch-project/sql/pull/3061
+        this.sourceBuilder.sort("_id", ASC);
```

Comment:
I see there are 3 places in the code where we sort on doc field.
Can you help me understand why we only need this here?

https://github.com/search?q=repo%3Aopensearch-project%2Fsql%20%22sort(DOC_FIELD_NAME%2C%20ASC)%22&type=code